app/vmalert/alerting.go

```diff
@@ -25,7 +25,7 @@ type AlertingRule struct {
 	For         time.Duration
 	Labels      map[string]string
 	Annotations map[string]string
-	GroupID     uint64
+	Group       *Group
```

Comment:
What do you think, is  it better to add new field GroupName instead of sharing mutex protected variable? Not sure, but potentially it may lead to data-races.

Reply:
Yeah, I noticed that and decided that Name is never accessed concurrently. But it doesn't stop to use other protected fields, so you're right.